pkg/lightning/backend/local/localhelper.go

```diff
@@ -172,10 +185,13 @@ func (s *storeWriteLimiter) WaitN(ctx context.Context, storeID uint64, n int) er
 }
 
 func (s *storeWriteLimiter) Limit() int {
-	return s.limit
+	return int(atomic.LoadInt64(&s.limit))
```

Comment:
make `s.limit` and `burst` a `atomic.Int64`

Reply:
Above update in https://github.com/pingcap/tidb/pull/57611/commits/fc28a05821bd83f791fbbb27b40b30d1df083aed